echo/README.md

```diff
@@ -1,5 +1,233 @@
+    # ...
+    ```
+
+8. Now it's time to run our actor:
+
+    ```bash
+    $ wasmcloud -m manifest.yaml
+    ```
+
+9. As a final step we will send another request to `http://localhost:8080`:
+
+    ```bash
+    curl http://localhost:8080/
+    ```
+
+## Using `wash` for better developer experience
+
+Previously we were using a declarative deployment model that did not allow us to modify resources at runtime. In this section, we will use `NATS` to allow `wasmcloud` to utilize self-managing [lattice network](https://wasmcloud.dev/reference/lattice/), and we can use `wash` to interact with the `wasmcloud` host via the lattice control interface (`ctl`).
+
+We will also use a local docker registry to store our echo actor to illustrate interacting with a local development version of an OCI registry.
+
+1. Since we are going to use `docker-compose` to run NATS server and  image registry, you need to have `wasmcloud`, `wash`, `docker` and `docker-compose` installed on your machine.
+
+2. You can either use `docker-compose.yml` file in the project's directory or create one with the following content:
+
+    ```yaml
+    version: "3"
+    services:
+      registry:
+        image: registry:2
+        ports:
+        - "5000:5000"
+      nats:
+        image: nats:2.1.9
+        ports:
+          - "6222:6222"
+          - "4222:4222"
+          - "8222:8222"
+    ```
+
+    We will run our containers in detached mode to reduce clutter.
+
+    ```bash
+    $ docker-compose up -d
+    ```
+
+    Start a REPL session using `wash up` command:
+
+    ```bash
+    $ wash up
+    ```
+
+    Check if session is running:
+
+    ```bash
+    $ ctl get hosts
+    ```
+
+    It will output something like:
+
+    ```bash
+    Host ID                                                   Uptime (seconds)
+    NDGPVEOZKAYT6JRL2C4BDVSW3PBBOYTWRLC2QFCSLXWDX4KCIRHM6WCZ  1357
+    ```
+
+3. Please open a new terminal, as the REPL is running in the current one. We will build our actor first, then will sign it:
+
+    ```bash
+    $ cargo build --release
+    $ wash claims sign ./target/wasm32-unknown-unknown/release/echo.wasm -c wasmcloud:httpserver --name "echo" --ver 0.2.2
+    ```
+
+    Or you can use `make` file in project's directory for convenience:
+
+    ```bash
+    make release VERSION=0.2.2
+    ```
+
+    Now we have our actor, `echo.wasm` and its signed version `echo_s.wasm` in `./target/wasm32-unknown-unknown/release` directory.
+
+3. Now it's time to push signed our actor into to local registry using `wash`:
+
+    ```bash
+    wash reg push localhost:5000/echo:0.2.2 ./target/wasm32-unknown-unknown/release/echo_s.wasm --insecure
+    ```
+
+4. Now, please go back to REPL terminal and run the actor from the local image registry:
+
+    ```bash
+    ctl start actor localhost:5000/echo:0.2.2
+    ```
+
+    Steps from now on is independent of our current example but we kept it for completeness:
```

Comment:
suggestion: The steps from now on are independent of our current example, but we've kept it for completeness